Pachinko/Source.c

```diff
@@ -0,0 +1,133 @@
+#include <stdio.h>
+#include <ctype.h>
+
+#define LINESIZE (1024)
+#define MINIMUM_CREDITS (100)
+#define BEGINNING_CREDITS (1000)
+#define NUMBER_OF_SYMBOLS (5)
+#define NUMBER_PER_LINE (3)
+#define JACKPOT_MULTIPLIER (3)
+
+int credits;
+int playedCredits;
+char symbols[NUMBER_OF_SYMBOLS];
+char spunSymbols[NUMBER_PER_LINE];
+
+
+void Init() 
+{
+	symbols[0] = '^';
+	symbols[1] = '%';
+	symbols[2] = 'X';
+	symbols[3] = '7';
+	symbols[4] = '*';
+	credits = BEGINNING_CREDITS;
+	playedCredits = 0;
+}
+
+void Spin()
+{
+	/* Use current time as seed for random generator */
+	srand((unsigned int)time(0));
+	spunSymbols[0] = symbols[rand() % NUMBER_OF_SYMBOLS];
+	spunSymbols[1] = symbols[rand() % NUMBER_OF_SYMBOLS];
+	spunSymbols[2] = symbols[rand() % NUMBER_OF_SYMBOLS];
+}
+
+void UpdateWin()
+{
+	credits -= playedCredits;
+	credits += playedCredits * JACKPOT_MULTIPLIER;
+}
+
+void UpdateLoss()
+{
+	credits -= playedCredits;
+}
+
+void RenderResults() 
+{
+	printf("\nSpinning...\n");
+	printf("%c %c %c\n", spunSymbols[0], spunSymbols[1], spunSymbols[2]);
+
+	if (spunSymbols[0] == spunSymbols[1] && spunSymbols[0] == spunSymbols[2])
+	{
+		printf("Jackpot! You won %d credits!\n\n", (playedCredits * JACKPOT_MULTIPLIER));
+	}
+	else
+	{
+		printf("You busted. You lost %d credits.\n\n", playedCredits);
+	}
+	if (credits < MINIMUM_CREDITS)
+	{
+		printf("You don't have enough credits to continue playing. Goodbye.\n");
+	}
+}
+
+void EvaluateResults() 
+{
+	if (spunSymbols[0] == spunSymbols[1] && spunSymbols[0] == spunSymbols[2]) 
+	{
+		UpdateWin();
```

Comment:
The reason I made so many functions is that you critiqued my code last time because I was 'renderring' in the same place I was updating or vice versa. So this aimed to remedy that (even though you said we don't need to use the game development structure). It was definitely cumbersome though, I agree, and the excess functions have been refactored.

Reply:
Yeah so you gotta decide if you will make something into a function or not, which takes time to get used.